### PR TITLE
Sexp2latex

### LINKED ISSUE #667: [CI] use local skema-rs REST API when testing Python-based workflows
To reliably test our system, we need to test our Python-based REST workflows against a local skema-rs REST API service.

### PR DESCRIPTION
**Pull request by ualiangzhang**
## Bug Fixes and Enhancements

- **img2mml Fixes:**
   - Addressed several bugs in img2mml, ensuring proper Unicode output.

- **Normalized Number Output:**
   - Merged normalized and processed numbers in the img2mml output for consistency.

- **to_latex Function Fixes:**
   - Resolved a series of bugs and issues in the to_latex function.

- [@myedibleenso] Fixes a bug in #668 to resolve #667

- [@myedibleenso] Pydantic v2 related changes to `skema/rest/llm_proxy.py` and `skema/skema_py/server.py`

## New `img2latex` Workflow

- **img2latex Implementation:**
   - Implemented a new img2latex workflow, integrating img2mml, mathml parser, and to_latex for seamless image-to-Latex conversion.

### CODE REVIEW

**Comment by ualiangzhang** [comment]
Hi, @myedibleenso! I have implemented a new workflow for img2latex. Could you please take a look? Thank you!

File: skema/rest/workflows.py [hunk 1]
```diff
@@ -66,6 +68,43 @@ async def equations_to_amr(data: schema.EquationImagesToAMR):
...
+    """
+    Converts images of equations to LaTeX.
+
+    ### Python example
+    ```
+    Endpoint for generating LaTeX from an input image.
+
+    ### Python example
+    ```
+    import requests
```
Comment (myedibleenso) [inline skema/rest/workflows.py:83]: @ualiangzhang , did you test to see if the include Python example here works with your new endpoint?

**Review by myedibleenso — CHANGES_REQUESTED** [review]
@ualiangzhang , can you add a test or two for some of these new functions?  Anything testing user-facing stuff (ex. your new endpoints) should be priortized.

File: skema/rest/workflows.py (hunk 1 above)
Comment (ualiangzhang) [reply skema/rest/workflows.py:83]: Yes. I tested it before. It worked for me.

File: skema/skema-rs/skema/src/services/mathml.rs [hunk 2]
```diff
@@ -60,6 +63,25 @@ pub async fn get_math_exp_graph(payload: String) -> String {
     dot_representation.to_string()
 }
 
+/// Parse a presentation MathML representation of an equation and
+/// return the corresponding LaTeX representation
+#[utoipa::path(
+request_body = String,
+responses(
```
Comment (adarshp) [inline skema/skema-rs/skema/src/services/mathml.rs:70]: Can you fix the indentation of the parentheses over here? It looks a bit strange?

**Review by adarshp — APPROVED** [review]
Just a small indentation-related comment. Otherwise everything looks good to me.

**Comment by myedibleenso** [comment]
I think you're already aware of this, @ualiangzhang , but your new test seems to be failing (`test_post_image_to_latex`): https://github.com/ml4ai/skema/actions/runs/6910133612/job/18802730906?pr=668#step:14:101

Let me know if you need help debugging.

**Comment by ualiangzhang** [comment]
Thanks, @myedibleenso ! Yes, I noticed that issue, and I am trying to fix that. I just don't know how to use TestClient to test the workflow properly, which always gets me a 404 error.

**Review by myedibleenso — COMMENTED** [review]
Thanks for adding that test, @ualiangzhang .  It's now passing in CI, but I think the expected response (according to the test) is not quite what we want (ex. surrounded in quotes, extra escapes, etc.).  Please confirm.

**Review by myedibleenso — APPROVED** [review]
Thanks, @ualiangzhang !

**Comment by ualiangzhang** [comment]
Thank you for your help, @myedibleenso !

**Pull request merged**